packages/terra-button/src/Button.jsx

```diff
@@ -127,6 +127,10 @@ class Button extends React.Component {
   }
 
   handleOnBlur(event) {
+    this.shouldShowFocus = false;
+    if (event.currentTarget.type === 'button') {
+      this.shouldShowFocus = true;
+    }
     this.setState({ focused: false });
```

Comment:
why is `shouldShowFocus` and the `focused` state different?

Reply:
Few things have changed here 746bdbe9423f725930e2d631bc5b336be1cd55a0. Let me know if you are still confused.